### PR TITLE
Refacto engine

### PR DESCRIPTION
**Pull request by bouda1**
contact and commands are C++ classes now.
The code is not finished but it is a base for next steps in this refacto.

### CODE REVIEW

File: src/objects/command.cc [hunk 1]
```diff
@@ -87,53 +87,52 @@ std::ostream& operator<<(std::ostream& os, command const& obj) {
...
+//      NEBATTR_NONE,
+//      obj.get(),
+//      &tv);
+//  }
+//  catch (...) {
+//    obj.reset();
+//  }
+//
+//  return obj.get();
+//}
```
Comment (SylvestreG) [inline src/objects/command.cc]: not needed anymore ? if yes you can delete this commented code.

File: src/compatibility/find.cc [hunk 2]
```diff
@@ -38,41 +38,23 @@ using namespace com::centreon;
...
+//commands::command* find_command(char const* name) {
+//  if (!name)
+//    return NULL;
+//
+//  std::unordered_map<std::string, std::shared_ptr<commands::command>>::const_iterator
+//    it(state::instance().commands().find(name));
+//  if (it != state::instance().commands().end())
+//    return it->second.get();
+//  return NULL;
+//}
```
Comment (SylvestreG) [inline src/compatibility/find.cc]: you can delete this commented code.

File: inc/com/centreon/engine/configuration/applier/state.hh [hunk 3]
```diff
@@ -75,115 +78,119 @@ namespace           configuration {
...
+      std::unordered_map<unsigned int, std::shared_ptr<host_struct>>&
                     hosts() throw ();
-      umap<unsigned int, std::shared_ptr<host_struct> >::const_iterator
+      std::unordered_map<unsigned int, std::shared_ptr<host_struct>>::const_iterator
                     hosts_find(configuration::host::key_type const& k) const;
-      umap<unsigned int, std::shared_ptr<host_struct> >::iterator
+      std::unordered_map<unsigned int, std::shared_ptr<host_struct>>::iterator
                     hosts_find(configuration::host::key_type const& k);
-      umultimap<std::string, std::shared_ptr<hostdependency_struct> > const&
+      umultimap<std::string, std::shared_ptr<hostdependency_struct>> const&
```
[the remark this answers is not in the document]
Comment (bouda1) [reply inc/com/centreon/engine/configuration/applier/state.hh]: Yes, I would prefer... :-)

File: src/configuration/applier/state.cc [hunk 4]
```diff
@@ -184,6 +186,7 @@ applier::state::state()
  *  Destructor.
  */
 applier::state::~state() throw() {
+  std::cout << "CONFIGURATION/APPLIER/STATE DESTRUCTOR...\n";
```
[the remark this answers is not in the document]
Comment (bouda1) [reply src/configuration/applier/state.cc]: Yep, my bad :-(

File: src/configuration/applier/state.cc [hunk 5]
```diff
@@ -196,143 +199,185 @@ applier::state::~state() throw() {
...
+std::unordered_map<std::string, std::shared_ptr<commands::command>>& applier::state::commands() throw () {
+  return _commands;
 }
 
 /**
- *  Find a command from its key.
+ *  Given a command name, find a command from the list in memory.
  *
- *  @param[in] k Command name.
+ *  @param[in] name Command name.
```
[the remark this answers is not in the document]
Comment (bouda1) [reply src/configuration/applier/state.cc]: done

File: src/configuration/applier/state.cc [hunk 6]
```diff
@@ -196,143 +199,185 @@ applier::state::~state() throw() {
...
+    return it->second.get();
+  return NULL;
 }
 
 /**
- *  Find a command from its key.
+ *  Given a command name, find a command from the list in memory.
  *
- *  @param[in] k Command name.
+ *  @param[in] name Command name.
```
[the remark this answers is not in the document]
Comment (bouda1) [reply src/configuration/applier/state.cc]: done

File: src/configuration/applier/state.cc [hunk 7]
```diff
@@ -196,143 +199,185 @@ applier::state::~state() throw() {
...
+    return it->second.get();
+  return NULL;
 }
 
 /**
- *  Find a connector from its key.
+ *  Given a contact name, find a contact from the list in memory.
  *
- *  @param[in] k Connector name.
+ *  @param[in] name Contact name.
```
[the remark this answers is not in the document]
Comment (bouda1) [reply src/configuration/applier/state.cc]: done

File: src/configuration/applier/state.cc [hunk 8]
```diff
@@ -196,143 +199,185 @@ applier::state::~state() throw() {
...
```
[the remark this answers is not in the document]
Comment (bouda1) [reply src/configuration/applier/state.cc]: done

File: src/configuration/applier/state.cc [hunk 9]
```diff
@@ -196,143 +199,185 @@ applier::state::~state() throw() {
...
+std::unordered_map<std::string, std::shared_ptr<commands::connector> >& applier::state::connectors() throw () {
+  return _connectors;
 }
 
 /**
- *  Get the current contacts.
+ *  Given a connector name, find a connector from the list in memory.
  *
- *  @return The current contacts.
+ *  @param[in] name Command name.
```
[the remark this answers is not in the document]
Comment (bouda1) [reply src/configuration/applier/state.cc]: done

File: src/configuration/applier/state.cc [hunk 10]
```diff
@@ -196,143 +199,185 @@ applier::state::~state() throw() {
...
+    return it->second.get();
+  return NULL;
 }
 
 /**
- *  Get the current contacts.
+ *  Given a connector name, find a connector from the list in memory.
  *
- *  @return The current contacts.
+ *  @param[in] name Command name.
```
[the remark this answers is not in the document]
Comment (bouda1) [reply src/configuration/applier/state.cc]: done

File: src/objects/command.cc [hunk 11]
```diff
@@ -143,13 +142,13 @@ command* add_command(char const* name, char const* value) {
...
-    throw (engine_error() << "Command '" << name << "' was not found");
-  return (*it->second);
-}
+//command& engine::find_command(std::string const& name) {
+//  umap<std::string, std::shared_ptr<command_struct> >::const_iterator
+//    it(state::instance().commands().find(name));
+//  if (it == state::instance().commands().end())
+//    throw (engine_error() << "Command '" << name << "' was not found");
+//  return *it->second;
+//}
```
[the remark this answers is not in the document]
Comment (bouda1) [reply src/objects/command.cc]: this file no more exists

File: src/objects/command.cc (hunk 1 above)
Comment (bouda1) [reply src/objects/command.cc]: the file is dead

File: src/compatibility/find.cc (hunk 2 above)
Comment (bouda1) [reply src/compatibility/find.cc]: done

**Pull request closed without merging**